firefox-ios/Providers/Pocket/PocketProvider.swift

```diff
@@ -66,49 +65,56 @@ class PocketProvider: PocketStoriesProviding, FeatureFlaggable, URLCaching {
     }
 
     // Fetch items from the global pocket feed
-    func fetchStories(items: Int, completion: @escaping (StoryResult) -> Void) {
-        let isFeatureEnabled = prefs.boolForKey(PrefsKeys.UserFeatureFlagPrefs.ASPocketStories) ?? true
-        let isCurrentLocaleSupported = PocketProvider.islocaleSupported(Locale.current.identifier)
+    func fetchStories(items: Int) async throws -> [PocketFeedStory] {
+        do {
+            let isFeatureEnabled = prefs.boolForKey(PrefsKeys.UserFeatureFlagPrefs.ASPocketStories) ?? true
+            let isCurrentLocaleSupported = PocketProvider.islocaleSupported(Locale.current.identifier)
 
-        // Check if we should use mock data
-        if shouldUseMockData {
-            return getMockDataFeed(count: items, completion: completion)
-        }
+            // Check if we should use mock data
+            if shouldUseMockData {
+                return try await getMockDataFeed(count: items)
+            }
 
-        // Ensure the feature is enabled and current locale is supported
-        guard isFeatureEnabled, isCurrentLocaleSupported else {
-            completion(.failure(Error.failure))
-            return
-        }
+            // Ensure the feature is enabled and current locale is supported
+            guard isFeatureEnabled, isCurrentLocaleSupported else {
+                throw Error.failure
+            }
 
-        // Note: Global feed is restricted to specific locale and feature availability
-        getGlobalFeed(items: items, completion: completion)
+            // Note: Global feed is restricted to specific locale and feature availability
+            return try await getGlobalFeed(items: items)
+        } catch {
+            throw error
+        }
```

Comment:
nit: This is a throwing method so you don't need the extra top level `do { ... } catch { throw error }`